src/Tickets/Emails/Dispatcher.php

```diff
@@ -0,0 +1,507 @@
+
+		/**
+		 * Allows modifications of the Email Dispatcher to all Email Types.
+		 *
+		 * @since TBD
+		 *
+		 * @param Dispatcher     $dispatcher Which dispatcher instance will be used for the email sent.
+		 * @param Email_Abstract $email      Which instance of the email that will be attached to this dispatcher.
+		 */
+		$dispatcher = apply_filters( 'tec_tickets_emails_dispatcher', $dispatcher, $email );
+
+		$email_slug = $email->slug;
+
+		/**
+		 * Allows modifications of the Email Dispatcher specific to this Email Type.
+		 *
+		 * @since TBD
+		 *
+		 * @param Dispatcher     $dispatcher Which dispatcher instance will be used for the email sent.
+		 * @param Email_Abstract $email      Which instance of the email that will be attached to this dispatcher.
+		 */
+		return apply_filters( "tec_tickets_emails_{$email_slug}_dispatcher", $dispatcher, $email );
+	}
+
+	/**
+	 * Includes a new header to be dispatched.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key
+	 * @param string $value
+	 */
+	public function add_header( string $key, string $value ): void {
+		$this->headers[ $key ] = $value;
+	}
+
+	/**
+	 * Includes a new attachment to be dispatched.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $path
+	 */
+	public function add_attachment( string $path ): void {
+		$this->attachments[] = $path;
+	}
+
+	/**
+	 * Get dispatcher headers.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_headers(): array {
+		/**
+		 * Filter the headers for all dispatchers.
+		 *
+		 * @since TBD
+		 *
+		 * @param array          $headers The headers.
+		 * @param Email_Abstract $this    The email object.
+		 */
+		$headers = apply_filters( 'tec_tickets_emails_dispatcher_headers', $this->headers, $this );
+
+		$email_slug = $this->get_email()->slug;
+
+		/**
+		 * Filter the headers for the particular email using this dispatcher.
+		 *
+		 * @since TBD
+		 *
+		 * @param array          $headers The headers.
+		 * @param Email_Abstract $this    The email object.
+		 */
+		$headers = apply_filters( "tec_tickets_emails_dispatcher_{$email_slug}_headers", $headers, $this );
+
+		// Enforce as an array
+		return (array) $headers;
+	}
+
+	/**
+	 * Get the formatted headers for this dispatcher.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_headers_formatted(): array {
+		$original_headers = $this->get_headers();
+		$headers          = [];
+
+		foreach ( $original_headers as $key => $value ) {
+			$headers[] = "{$key}: {$value}";
+		}
+
+		return $headers;
+	}
+
+	/**
+	 * Get email attachments.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_attachments(): array {
+		/**
+		 * Filter the attachments.
+		 *
+		 * @since TBD
+		 *
+		 * @param array          $attachments The attachments.
+		 * @param Email_Abstract $this        The email object.
+		 */
+		$attachments = apply_filters( 'tec_tickets_emails_dispatcher_attachments', $this->attachments, $this );
```

Comment:
Why don't you keep the attachment method in the email class (where it belongs, because it's tied to emails) and grab the headers from there by using the `$this->get_email()->get_attachments()`?

Naming this with `dispatcher` adds more noise, making it more difficult to access the actual email if you're trying (as a dev extending the functionality) to access the actual email.

⚠️ **This particular change breaks the TEC implementation**. We should be more careful with these things. There's a lack of context, and we're introducing breaking changes in the week that we should mark this feature as ready.